src/data/finite/basic.lean

```diff
@@ -62,33 +43,12 @@ open_locale classical
 
 variables {α β γ : Type*}
 
-/-- A type is `finite` if it is in bijective correspondence to some
-`fin n`.
-
-While this could be defined as `nonempty (fintype α)`, it is defined
-in this way to allow there to be `finite` instances for propositions.
--/
-class inductive finite (α : Sort*) : Prop
-| intro {n : ℕ} : α ≃ fin n → finite
-
-lemma finite_iff_exists_equiv_fin {α : Sort*} : finite α ↔ ∃ n, nonempty (α ≃ fin n) :=
-⟨λ ⟨e⟩, ⟨_, ⟨e⟩⟩, λ ⟨n, ⟨e⟩⟩, ⟨e⟩⟩
-
-lemma finite.exists_equiv_fin (α : Sort*) [h : finite α] : ∃ (n : ℕ), nonempty (α ≃ fin n) :=
-finite_iff_exists_equiv_fin.mp h
-
-lemma finite.of_equiv (α : Sort*) {β : Sort*} [h : finite α] (f : α ≃ β) : finite β :=
-by { casesI h with n e, exact finite.intro (f.symm.trans e) }
-
-lemma equiv.finite_iff {α β : Sort*} (f : α ≃ β) : finite α ↔ finite β :=
-⟨λ _, by exactI finite.of_equiv _ f, λ _, by exactI finite.of_equiv _ f.symm⟩
-
-lemma finite.of_fintype {α : Type*} (h : fintype α) : finite α := ⟨fintype.equiv_fin α⟩
+protected lemma fintype.finite {α : Type*} (h : fintype α) : finite α := ⟨fintype.equiv_fin α⟩
```

Comment:
Should this be `fintype.to_finite`, or do you think the name is fine? (Note that the comment in `data/finite/defs` with the typo doesn't match.)  I'm fine with either.

Reply:
We have quite a few dot notation lemmas without `to_`. Some day, we should decide about some coding style here and enforce it.